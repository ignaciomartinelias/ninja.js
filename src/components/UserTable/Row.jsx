import React from 'react'

const UserRow = ({ row }) => {
  return (
    <tr>
      <td>
        <a href={row.website}>
          {row.name}
        </a><br />
        <small>{row.email}</small>
      </td>
    </tr>
  )
}

export default UserRow
