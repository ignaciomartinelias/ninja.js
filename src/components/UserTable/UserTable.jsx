import React, { useEffect } from "react";
import { useSelector } from "react-redux";
import { fetchUserList } from "../../actions/userActions";
import store from "../../store";
import DataTable from "../DataTable/DataTable.jsx";
import Row from "./Row";

const UserTable = () => {
  const users = useSelector(state => state.userReducer.users);

  const handleSearch = (users, text) => {
    return users.filter(
      user =>
        user.name1.toLowerCase().includes(text.toLowerCase()) || user.email.toLowerCase().includes(text.toLowerCase())
    );
  };

  useEffect(() => {
    store.dispatch(fetchUserList());
  }, []);

  return <DataTable rows={users} rowsPerPage={5} handleSearch={handleSearch} RowToRender={Row} />;
};

export default UserTable;
