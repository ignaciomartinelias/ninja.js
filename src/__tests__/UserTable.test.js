import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import toJson from "enzyme-to-json";
import UserTable from "../components/UserTable/UserTable";

const mockStore = configureStore();
const store = mockStore({
  dataTableReducer: {
    currentPageNumber: 0
  },
  userReducer: {
    users: [
      {
        name: "Mads L. Klausen",
        email: "MadsLKlausen@jourrapide.com",
        edit_path: "http://google.com",
        id: 1,
      },
      {
        name: "Alfred K. Krogh",
        email: "AlfredKKrogh@armyspy.com",
        edit_path: "http://google.com",
        id: 2,
      },
      {
        name: "Silas L. Bertelsen",
        email: "SilasLBertelsen@armyspy.com",
        edit_path: "http://google.com",
        id: 3,
      },
      {
        name: "Mia A. Johnsen",
        email: "MiaAJohnsen@dayrep.com",
        edit_path: "http://google.com",
        id: 4,
      },
      {
        name: "Alfred S. Schou",
        email: "AlfredSSchou@jourrapide.com",
        edit_path: "http://google.com",
        id: 5,
      },
      {
        name: "Mads L. Klausen",
        email: "MadsLKlausen@jourrapide.com",
        edit_path: "http://google.com",
        id: 6,
      },
      {
        name: "Alfred K. Krogh",
        email: "AlfredKKrogh@armyspy.com",
        edit_path: "http://google.com",
        id: 7,
      },
      {
        name: "Silas L. Bertelsen",
        email: "SilasLBertelsen@armyspy.com",
        edit_path: "http://google.com",
        id: 8,
      },
      {
        name: "Mia A. Johnsen",
        email: "MiaAJohnsen@dayrep.com",
        edit_path: "http://google.com",
        id: 9,
      },
      {
        name: "Alfred S. Schou",
        email: "AlfredSSchou@jourrapide.com",
        edit_path: "http://google.com",
        id: 10,
      },
    ],
  },
});

it("renders 4 rows", () => {
  const wrapper = mount(
    <Provider store={store}>
      <UserTable />
    </Provider>
  );
  expect(wrapper.find("tr").length).toBe(4);
});

it("filters rows based on input", () => {
  const wrapper = mount(
    <Provider store={store}>
      <UserTable />
    </Provider>
  );

  wrapper.find("input").simulate("change", { target: { value: "k" } });
  expect(wrapper.find("tr").length).toBe(4);
});

it("generates 2 pages and first page is active", () => {
  const wrapper = mount(
    <Provider store={store}>
      <UserTable />
    </Provider>
  );
  const pageButtons = wrapper.find("Pagination Page");
  expect(pageButtons.length).toBe(4); // 2 per page plus the next button
  expect(pageButtons.at(0).find("button").prop("className")).toContain("button-outline");
});

it("Snapshot UserTable Component", () => {
  const wrapper = mount(
    <Provider store={store}>
      <UserTable />
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
