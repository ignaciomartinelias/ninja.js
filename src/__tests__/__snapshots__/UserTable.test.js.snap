// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot UserTable Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UserTable>
    <DataTable
      RowToRender={[Function]}
      handleSearch={[Function]}
      rows={
        Array [
          Object {
            "edit_path": "http://google.com",
            "email": "MadsLKlausen@jourrapide.com",
            "id": 1,
            "name": "Mads L. Klausen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "AlfredKKrogh@armyspy.com",
            "id": 2,
            "name": "Alfred K. Krogh",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "SilasLBertelsen@armyspy.com",
            "id": 3,
            "name": "Silas L. Bertelsen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "MiaAJohnsen@dayrep.com",
            "id": 4,
            "name": "Mia A. Johnsen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "AlfredSSchou@jourrapide.com",
            "id": 5,
            "name": "Alfred S. Schou",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "MadsLKlausen@jourrapide.com",
            "id": 6,
            "name": "Mads L. Klausen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "AlfredKKrogh@armyspy.com",
            "id": 7,
            "name": "Alfred K. Krogh",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "SilasLBertelsen@armyspy.com",
            "id": 8,
            "name": "Silas L. Bertelsen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "MiaAJohnsen@dayrep.com",
            "id": 9,
            "name": "Mia A. Johnsen",
          },
          Object {
            "edit_path": "http://google.com",
            "email": "AlfredSSchou@jourrapide.com",
            "id": 10,
            "name": "Alfred S. Schou",
          },
        ]
      }
      rowsPerPage={4}
    >
      <div>
        <Search
          onSearch={[Function]}
        >
          <div
            className="p-b-1"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Search"
              type="search"
            />
          </div>
        </Search>
        <table>
          <tbody>
            <UserRow
              key="1"
              row={
                Object {
                  "edit_path": "http://google.com",
                  "email": "MadsLKlausen@jourrapide.com",
                  "id": 1,
                  "name": "Mads L. Klausen",
                }
              }
            >
              <tr>
                <td>
                  <a>
                    Mads L. Klausen
                  </a>
                  <br />
                  <small>
                    MadsLKlausen@jourrapide.com
                  </small>
                </td>
              </tr>
            </UserRow>
            <UserRow
              key="2"
              row={
                Object {
                  "edit_path": "http://google.com",
                  "email": "AlfredKKrogh@armyspy.com",
                  "id": 2,
                  "name": "Alfred K. Krogh",
                }
              }
            >
              <tr>
                <td>
                  <a>
                    Alfred K. Krogh
                  </a>
                  <br />
                  <small>
                    AlfredKKrogh@armyspy.com
                  </small>
                </td>
              </tr>
            </UserRow>
            <UserRow
              key="3"
              row={
                Object {
                  "edit_path": "http://google.com",
                  "email": "SilasLBertelsen@armyspy.com",
                  "id": 3,
                  "name": "Silas L. Bertelsen",
                }
              }
            >
              <tr>
                <td>
                  <a>
                    Silas L. Bertelsen
                  </a>
                  <br />
                  <small>
                    SilasLBertelsen@armyspy.com
                  </small>
                </td>
              </tr>
            </UserRow>
            <UserRow
              key="4"
              row={
                Object {
                  "edit_path": "http://google.com",
                  "email": "MiaAJohnsen@dayrep.com",
                  "id": 4,
                  "name": "Mia A. Johnsen",
                }
              }
            >
              <tr>
                <td>
                  <a>
                    Mia A. Johnsen
                  </a>
                  <br />
                  <small>
                    MiaAJohnsen@dayrep.com
                  </small>
                </td>
              </tr>
            </UserRow>
          </tbody>
        </table>
        <Pagination
          currentPageNumber={0}
          numberOfPages={3}
        >
          <ul
            className="pagination"
          >
            <Page
              key="0"
              pageNumber={0}
            >
              <li
                className="page-item mr-1"
              >
                <button
                  className="page-link button-outline"
                  onClick={[Function]}
                >
                  1
                </button>
              </li>
            </Page>
            <Page
              key="1"
              pageNumber={1}
            >
              <li
                className="page-item mr-1"
              >
                <button
                  className="page-link "
                  onClick={[Function]}
                >
                  2
                </button>
              </li>
            </Page>
            <Page
              key="2"
              pageNumber={2}
            >
              <li
                className="page-item mr-1"
              >
                <button
                  className="page-link "
                  onClick={[Function]}
                >
                  3
                </button>
              </li>
            </Page>
            <Page
              pageNumber={1}
              text=">"
            >
              <li
                className="page-item mr-1"
              >
                <button
                  className="page-link "
                  onClick={[Function]}
                >
                  &gt;
                </button>
              </li>
            </Page>
          </ul>
        </Pagination>
      </div>
    </DataTable>
  </UserTable>
</Provider>
`;
